frame/staking/src/lib.rs

```diff
@@ -1465,81 +1863,108 @@ impl<T: Trait> Module<T> {
 		})
 	}
 
-	/// Select a new validator set from the assembled stakers and their role preferences.
-	///
-	/// Returns the new `SlotStake` value and a set of newly selected _stash_ IDs.
+	/// Runs [`try_do_phragmen`] and updates the following storage items:
+	/// - [`Stakers`]
+	/// - [`SlotStake`]
+	/// - [`CurrentElected]
 	///
-	/// Assumes storage is coherent with the declaration.
-	fn select_validators() -> (BalanceOf<T>, Option<Vec<T::AccountId>>) {
-		let mut all_nominators: Vec<(T::AccountId, Vec<T::AccountId>)> = Vec::new();
-		let all_validator_candidates_iter = <Validators<T>>::enumerate();
-		let all_validators = all_validator_candidates_iter.map(|(who, _pref)| {
-			let self_vote = (who.clone(), vec![who.clone()]);
-			all_nominators.push(self_vote);
-			who
-		}).collect::<Vec<T::AccountId>>();
+	/// If the election has been successful. It passes the new set upwards.
+	fn select_and_update_validators() -> Option<Vec<T::AccountId>> {
+		if let Some(ElectionResult::<T::AccountId, BalanceOf<T>> {
+			elected_stashes,
+			exposures,
+			slot_stake,
+			compute,
+			.. // TODO: this means that we are never storing the score of a winning chain. It this okay?
```

Comment:
This is kinda worth noting. What do we want to know on chain at each era? ATM we store nothing. Only during the open window, the `QueuedElected` might have some value which is then `take()`n  and then we are back to the old staking with the same storage item. Maybe we want to store an instance of `ElectionResult` from now on to know each result came from where? with what score? who signed it? or was it unsigned? Atm we can just know this stuff by tracking events and logs.